src/transformStateWithClones.js

```
@@ -7,7 +7,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state }; // клон начального состояния
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = addProperties(currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        currentState = removeProperties(currentState, action.keysToRemove);
+        break;
+
+      case 'clear':
+        currentState = clearProperties(state);
+        break;
+    }
```

According to the provided checklist, a `switch` statement should have a `default` case to handle unexpected action types. This can prevent errors if an action with an unknown `type` is passed. For example, you could throw an error or simply do nothing.